### ISSUE TITLE
Error: Unable to retrieve bucket contents.

### ISSUE THREAD

**Issue by lBroth** [tylerhall/Autosmush#12]
Hi, 
I have an
Error: Unable to retrieve bucket contents. on ./autosmush subdomain.domain.com

The name of bucket is a subdomain because i have enabled website hosting for get images by subdomain.domain.com

Thanks

**Comment by tylerhall** [comment]
Sorry, I have no way of helping without further error logs.   

On May 10, 2014 at 8:43:50 PM CDT, lBroth notifications@github.com wrote:Hi,  I have an Error: Unable to retrieve bucket contents. on ./autosmush subdomain.domain.com  The name of bucket is a subdomain because i have enabled website hosting for get images by subdomain.domain.com  Thanks  —Reply to this email directly or view it on GitHub.

**Comment by raeesaa** [comment]
@lBroth : Were you able to resolve the error? Even I am getting same error. After digging into autosmush.php, I realized that Amazon was returning status 301 instead of 200 and therefore this error is getting logged.

**Comment by lBroth** [comment]
No, i don't use Autosmush on my bucket. Unfortunately with subdomain autosmush not get the bucket.
I solved it by resizing and compressing the image on my server and later put in amazon bucket.

**Comment by raeesaa** [comment]
That is shocking. I thought autosmush supports subdomain. 

Thanks for the reply.

**Comment by theoc31** [comment]
@tylerhall How can I get you the error logs or what do you need as I am also experiencing the same issue and get the error "Unable to retrieve bucket contents".

 My bucket is in the form of "subdomain.domain.com".
Thanks.

**Comment by theoc31** [comment]
Just an update for anyone intrigued.
I've realised that the reason why "Unable to retrieve bucket contents" appears is because, the HTTP status code returned is 301 rather than the needed 200.

If I bypass this check, when it retrieves the bucket contents through the AWS API, it returns 0 and the script again fails. Anyone have any ideas?

**Comment by iit2011081** [comment]
I am also facing same error  because bucket headers are returning 403 status code.  Is there any settings or configurations to be done in s3 bucket ?
